internal/tools/aws/database_multitenant.go

```diff
@@ -58,50 +58,34 @@ func NewRDSMultitenantDatabase(instanceID, installationID string, client *Client
 
 // Teardown removes all AWS resources related to a RDS multitenant database.
 func (d *RDSMultitenantDatabase) Teardown(store model.InstallationDatabaseStoreInterface, keepData bool, logger log.FieldLogger) error {
-	databaseName := MattermostRDSDatabaseName(d.installationID)
-	logger = logger.WithField("rds-multitenant-database", databaseName)
+	logger = logger.WithField("rds-multitenant-database", MattermostRDSDatabaseName(d.installationID))
+
+	logger.Info("Tearing down RDS multitenant database")
 
-	logger.Info("Tearing down RDS database and database secret")
+	if keepData {
+		logger.Warn("Keepdata is set to true on this server, but this is not yet supported for RDS multitenant databases")
+	}
 
 	multitenantDatabases, err := store.GetMultitenantDatabases(&model.MultitenantDatabaseFilter{
 		InstallationID:          d.installationID,
 		NumOfInstallationsLimit: model.NoInstallationsLimit,
 		PerPage:                 model.AllPerPage,
 	})
 	if err != nil {
-		return errors.Wrapf(err, "failed to find multitenant database for installation ID %s", d.installationID)
+		return errors.Wrap(err, "failed to query for multitenant databases")
 	}
 
-	if len(multitenantDatabases) > 1 {
-		return errors.Errorf("teardown expected exactly one multitenant database for installation ID %s (found %d)", d.installationID, len(multitenantDatabases))
-	}
-
-	if len(multitenantDatabases) < 1 {
-		logger.Infof("No multitenant database found for installation ID %s: teardown completed.", d.installationID)
-		return nil
-	}
-
-	unlocked, err := d.lockMultitenantDatabase(multitenantDatabases[0].ID, store)
-	if err != nil {
-		return errors.Wrapf(err, "failed to lock multitenant database ID %s when tearing the multitenant database down", multitenantDatabases[0].ID)
-	}
-	defer unlocked(logger)
-
-	rdsCluster, err := d.describeRDSCluster(multitenantDatabases[0].ID)
-	if err != nil {
-		return errors.Wrapf(err, "failed to describe multitenant database ID %s when tearing the multitenant database down", multitenantDatabases[0].ID)
-	}
-
-	logger = logger.WithField("rds-cluster-id", *rdsCluster.DBClusterIdentifier)
-
-	err = d.dropDatabaseAndDeleteSecret(multitenantDatabases[0].ID, *rdsCluster.Endpoint, databaseName, store, logger)
-	if err != nil {
-		return errors.Wrap(err, "unable to finish teardown of multitenant database")
-	}
-
-	err = d.updateTagCounterAndRemoveInstallationID(rdsCluster.DBClusterArn, multitenantDatabases[0], store, logger)
-	if err != nil {
-		return errors.Wrap(err, "unable to finish teardown of multitenant database")
+	numDatabases := len(multitenantDatabases)
+	switch {
+	case (numDatabases > 1):
+		return errors.Errorf("expected 1 or less multitenant databases from query, but found %d", len(multitenantDatabases))
```

Comment:
nit: Gabe, is there a reason you're returning `len(multitenantDatabases)` and not the `numDatabases` itself?

Reply:
Nope, there isn't! Just a copy-pasta bug. Thanks for catching it.